### DISCUSSION TITLE
Group by package in Monorepo

### DISCUSSION THREAD

**Discussion by estahn** [renovatebot/renovate#8726] [Request Help]
We have a repository `js-packages` containing multiple nodejs packages. This is managed by lerna etc. I'm trying to enable renovatebot to group updates by each package, possibly by package + dependency.

```
.
├── package.json
├── packages
│   ├── eslint
│   │   └── package.json
│   ├── file-upload
│   │   ├── package.json
│   ├── header
│   │   ├── package.json
│   ├── lint
│   │   ├── package.json
│   ├── nest
│   │   ├── package.json
│   ├── sd
│   │   ├── package.json
│   ├── select
│   │   ├── package.json
│   ├── ui
│   │   ├── package.json
│   └── with-login
│       ├── package.json

```

I have tried various combination without success. The config looks currently like this:
```json
{
  "automerge": false,
  "stabilityDays": 3,
  "timezone": "Australia/Sydney",
  "ignorePaths": [],
  "schedule": [
    "every weekday after 10am",
    "every weekday before 4pm"
  ],
  "extends": [
    "config:js-lib"
  ],
  "packageRules": [
    {
      "matchPaths": ["package.json"]
    }
  ]
}
```

**Comment by rarkins** [comment]
FYI the package rule you have does nothing and should be removed. 

If you want to separate PRs then you need to separate branches. You can do that this way:

```
  "additionalBranchPrefix": ["{{parentDir}}-"],
  "commitMessageSuffix": "({{parentDir}})"
```

In your example the `parentDir` is `eslint`, `file-upload`, etc. The `commitMessageSuffix` part is optional but there so that you can differentiate the same updates between different monorepos in the PR title and commit messages.
